snapcraft/tests/test_lxd.py

```diff
@@ -186,3 +191,148 @@ def test_remote_does_not_exist(self, mock_run):
         with ExpectedException(ContainerConnectionError,
                                'There are either.*{}.*'.format(self.remote)):
             self.make_cleanbuilder()
+
+    def test_parallel_invocation(self):
+        builder1 = self.make_cleanbuilder()
+        builder2 = self.make_cleanbuilder()
+        builder1.execute()
+        builder2.execute()
+
+    @patch('snapcraft.internal.common.is_snap')
+    def test_parallel_invocation_inject_snap(self, mock_is_snap):
+        mock_is_snap.side_effect = lambda: True
+
+        fake_snapd = tests.fixture_setup.FakeSnapd()
+        self.useFixture(fake_snapd)
+
+        builder1 = self.make_cleanbuilder()
+        builder2 = self.make_cleanbuilder()
+        builder1.execute()
+        builder2.execute()
+
```

Comment:
Maybe, maybe add the adapted assertion `self.fake_filesystem.rmtree_mock.assert_has_calls([call(builder.tmp)])` for each cleanbuilder instance at the end